code/crates/engine/src/consensus.rs

```diff
@@ -1047,7 +1046,7 @@ where
                 Ok(r.resume_with(()))
             }
 
-            Effect::GetVoteSet(height, round, r) => {
+            Effect::RequestVoteSet(height, round, r) => {
```

Comment:
Before this PR, with:
```
[sync]
enabled = false
```
the `rebroadcast` was taking effect, otherwise the `request-response`.
After this PR, with:
```
[consensus.vote_sync]
mode = "request-response"
...
[sync]
enabled = false
```
consensus will halt. It is a misconfiguration and we should maybe detect this earlier or at least log an error here if `self.sync.is_none()`.

Reply:
How about we change the `sync` section to be named `value_sync`, and spawn the `Sync` actor when either `value_sync` is true or `consensus.vote_sync.mode = "request-response"`?

Or maybe we should split the Sync actor into a ValueSync and VoteSync actor. Is that something you had started working on or not, I can't rembember.

Reply:
Wondering if we should we have `enabled` for `vote_sync` as well. I lean against it tbh since consensus will get stuck under some conditions if there is no sync at that level. And it's not easy or expected that an app will develop their own (like in the case of value_sync).

Reply:
Agreed, let's enforce VoteSync for now.